File: CS/ImageMemoEdit/CustomGridControl/CustomGridControl.cs
using System;
using System.Collections.Generic;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using System.ComponentModel;

namespace ImageTextEdit
{
    [System.ComponentModel.DesignerCategory("")]
    [System.ComponentModel.ToolboxItem(true)]
    public class CustomGridControl : GridControl
    {
        public CustomGridControl() : base() { }

        protected override void RegisterAvailableViewsCore(InfoCollection collection)
        {
            base.RegisterAvailableViewsCore(collection);
            collection.Add(new CustomGridInfoRegistrator());
        }

        protected override BaseView CreateDefaultView()
        {
            return CreateView("CustomGridView");
        }
    }
    public delegate void OnSmartIconSelectionEventHandler(object sender, SmartIconSelectionEventArgs e);
}

File: VB/ImageMemoEdit/CustomGridControl/CustomGridControl.vb
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Registrator
Imports System.ComponentModel

Namespace ImageTextEdit

    <System.ComponentModel.DesignerCategory("")>
    <System.ComponentModel.ToolboxItem(True)>
    Public Class CustomGridControl
        Inherits GridControl

        Public Sub New()
            MyBase.New()
        End Sub

        Protected Overrides Sub RegisterAvailableViewsCore(ByVal collection As InfoCollection)
            MyBase.RegisterAvailableViewsCore(collection)
            collection.Add(New CustomGridInfoRegistrator())
        End Sub

        Protected Overrides Function CreateDefaultView() As BaseView
            Return CreateView("CustomGridView")
        End Function
    End Class

    Public Delegate Sub OnSmartIconSelectionEventHandler(ByVal sender As Object, ByVal e As SmartIconSelectionEventArgs)
End Namespace
